core-sitemaps.php

```diff
@@ -17,4 +17,50 @@
  * @package         Core_Sitemaps
  */
 
-// Your code starts here.
+defined( 'ABSPATH' ) || die();
+
+const CORE_SITEMAPS_CPT_BUCKET       = 'core_sitemaps_bucket';
+const CORE_SITEMAPS_POSTS_PER_BUCKET = 2000;
+
+require_once __DIR__ . '/inc/bucket.php';
+require_once __DIR__ . '/inc/page.php';
+require_once __DIR__ . '/inc/type-post.php';
+require_once __DIR__ . '/inc/url.php';
+
+/**
+ * Bootstrapping.
+ */
+function core_sitemaps_init() {
+	// Fixme: temporarily unhooking template.
+	core_sitemaps_bucket_register();
+
+	$register_post_types = core_sitemaps_registered_post_types();
+	foreach ( array_keys( $register_post_types ) as $post_type ) {
+		call_user_func( $register_post_types[ $post_type ] );
+	}
+}
+
+add_action( 'init', 'core_sitemaps_init', 10 );
+
+/**
+ * Provides the `core_sitemaps_register_post_types` filter to register post types for inclusion in the sitemap.
+ *
+ * @return array Associative array.  Key is the post-type name; Value is a registration callback function.
+ */
+function core_sitemaps_registered_post_types() {
```

Comment:
This is an ok starting point, but I think that eventually an object oriented approach that uses a simplified [Registry pattern](https://designpatternsphp.readthedocs.io/en/latest/Structural/Registry/README.html) could be really useful here so we could register different Sitemap classes for each object type. [Yoast does something similar here](https://github.com/Yoast/wordpress-seo/blob/trunk/inc/sitemaps/class-sitemaps.php#L122-L142) which is a nice approach.